web/views/rights.py

```diff
@@ -40,4 +40,48 @@ def post(self):
         return res
 
 
+class RightTransferApi(Resource):
+    def post(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('rightId', type=right_model, required=True,
+                            location='json')
+        parser.add_argument('currentHolder', type=user_model, required=True,
+                            location='json')
+        parser.add_argument('to', type=user_model, required=True,
+                            location='json')
+        parser.add_argument('rightsAssignment', type=dict, location='json')
+        args = parser.parse_args()
+
+        right_id = args['rightId']
+        current_holder = args['currentHolder']
+        to = args['to']
+        rights_assignment = args['rights_assignment']
+
+        for user in [current_holder, to]:
+            user['verifying_key'] = user.pop('verifyingKey')
+            user['signing_key'] = user.pop('signingKey')
+
+        # We can't be sure of the type of Right that's given by using just the
+        # id, so let's assume it's a normal Right first before trying to make a
+        # Copyright
+        try:
+            right = models.Right.from_persist_id(right_id,
+                                                 plugin=coalaip.plugin,
+                                                 force_load=True)
+        except ModelDataError:
+            right = models.Copyright.from_persist_id(right_id,
+                                                     plugin=coalaip.plugin,
+                                                     force_load=True)
```

Comment:
This isn't super nice with the way I've set up Rights and Copyrights in `pycoalaip`, but it's either something like this where we do a best-effort guess or ask the user to supply the type of the Right.

The main issue is that, at the end of they day, we can never be 100% sure what an entity's type is if they haven't explicitly stated it themselves (`@type` breaks due to Manifestations able to be any submodel of CreativeWork).